modules/core/src/main/scala/fs2/kafka/Deserializer.scala

```diff
@@ -16,7 +16,7 @@ import java.util.UUID
   * Functional composable Kafka key- and record deserializer with
   * support for effect types.
   */
-sealed abstract class Deserializer[F[_], A] {
+sealed abstract class GenDeserializer[-T <: KeyOrValue, F[_], A] {
```

Comment:
What is `Gen`? Generic?

Reply:
I'm not actually sure - I'm following the pattern of things like `GenSpawn` in cats-effect. Either generic or generalized I guess...

Reply:
Maybe `GenericDeserializer` would be better? I believe that self-descriprive naming is more important than saving 4 letters to type.

Reply:
Yeah that's fair 😅